realm/realm-library/src/objectServer/java/io/realm/RealmApp.java

```diff
@@ -0,0 +1,413 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm;
+
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.Future;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import javax.annotation.Nullable;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.realm.internal.RealmNotifier;
+import io.realm.internal.android.AndroidCapabilities;
+import io.realm.internal.android.AndroidRealmNotifier;
+import io.realm.internal.async.RealmAsyncTaskImpl;
+import io.realm.internal.network.OkHttpNetworkTransport;
+import io.realm.internal.objectstore.OsJavaNetworkTransport;
+import io.realm.log.RealmLog;
+import io.realm.mongodb.RealmMongoDBService;
+
+/**
+ * FIXME
+ */
+public class RealmApp {
+
+    // Implementation notes:
+    // The public API's currently only allow for one RealmApp, however this is a restriction
+    // we might want to lift in the future. So any implementation details so ideally be made
+    // with that in mind, i.e. keep static state to minimum.
+
+    // Default session error handler handler that just output errors to LogCat
+    private static final SyncSession.ErrorHandler SESSION_NO_OP_ERROR_HANDLER = new SyncSession.ErrorHandler() {
+        @Override
+        public void onError(SyncSession session, ObjectServerError error) {
+            if (error.getErrorCode() == ErrorCode.CLIENT_RESET) {
```

Comment:
Client Reset support is still a bit unclear, but if anything has changed I would rather remove/make changes in a single PR since a lot of places are affected. This code was just a copy from `SyncManager`